pkg/certificate/signing_requestor.go

```diff
@@ -308,3 +379,72 @@ func deleteIfPresent(ctx context.Context, client dynamic.ResourceInterface, name
 
 	return errors.Wrapf(err, "error deleting Secret %q", name)
 }
+
+// startCertificateRenewalMonitoring starts periodic monitoring of certificate expiration.
+func (s *signingRequestorImpl) startCertificateRenewalMonitoring(stopCh <-chan struct{}) {
+	s.renewalTicker = time.NewTicker(CertCheckInterval)
+
+	wait.Until(s.checkCertificateRenewal, CertCheckInterval, stopCh)
+
+	logger.Info("Started certificate renewal monitoring", "interval", CertCheckInterval)
+}
+
+// updateCertificateExpiration extracts and stores the certificate expiration time.
+func (s *signingRequestorImpl) updateCertificateExpiration(secret *corev1.Secret) error {
+	certPEM := secret.Data[TLSDataKey]
+	if len(certPEM) == 0 {
+		return errors.Errorf("Failed to find certificate PEM in %q for secret %q", TLSDataKey, secret.Name)
+	}
+
+	block, _ := pem.Decode(certPEM)
+	if block == nil {
+		return errors.Errorf("Failed to find certificate PEM in %q for secret %q", TLSDataKey, secret.Name)
```

Comment:
Errors shouldn't be capitalized. Same below.

```suggestion
		return errors.Errorf("failed to find certificate PEM in %q for secret %q", TLSDataKey, secret.Name)
```